cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -469,6 +483,80 @@ func (o *objectGraph) Discovery(ctx context.Context, namespace string) error {
 				return errors.Wrapf(err, "failed to add obj (Kind=%s, Name=%s) to graph", obj.GetKind(), obj.GetName())
 			}
 		}
+
+		if typeMeta.GroupVersionKind().GroupKind() != clusterv1.GroupVersion.WithKind("Cluster").GroupKind() {
+			continue
+		}
```

Comment:
What about dropping this if and moving the next block of code outside after `for _, discoveryType := range o.types {}`, currently L561